crates/oxc_semantic/src/scope/builder.rs

```diff
@@ -104,6 +104,11 @@ impl ScopeBuilder {
             AstKind::ArrowExpression(_) => Some(ScopeFlags::Function | ScopeFlags::Arrow),
             AstKind::StaticBlock(_) => Some(ScopeFlags::ClassStaticBlock),
             AstKind::TSModuleBlock(_) => Some(ScopeFlags::TsModuleBlock),
+            AstKind::Class(class) if matches!(class.r#type, ClassType::ClassExpression) => {
+                // Class expression creates a temporary scope with the class name as its only variable
+                // E.g., `let c = class A { foo() { console.log(A) } }`
+                Some(ScopeFlags::empty())
+            }
```

Comment:
I need to lookup the spec for this one.

Reply:
This is an interesting case where TypeScript does a huge routine for this.

`c` gets the `Class` symbol while `A` gets an anonymous symbol ... it's really complicated. But your solution is fine here (for the foreseeable future.)